### PR TITLE
Expose Type.GetTypeCode() and Type.Delimiter

### PR DESCRIPTION
**Pull request by pallavit**

### CODE REVIEW

**Comment by pallavit** [comment]
/cc: @weshaggard Could you please review the change.

File: src/System.Runtime/tests/System.Runtime.Tests.csproj [hunk 1]
```diff
@@ -119,7 +119,17 @@
   </ItemGroup>
   <ItemGroup>
     <!-- Compile tests against the System.Runtime contract, but copy our local-built implementation for testing -->
+	<ProjectReference Include="..\ref\System.Runtime.csproj">
```
Comment (weshaggard) [inline src/System.Runtime/tests/System.Runtime.Tests.csproj]: This seems to indicate we are not testing any of the other new System.Runtime APIs higher then 4.0.20?

File: src/System.Runtime/tests/System.Runtime.Tests.csproj [hunk 2]
```diff
@@ -119,7 +119,17 @@
...
+      <Project>{446de6f0-47a0-4c2f-a918-768a76ccf461}</Project>
+      <Name>System.Runtime</Name>
+      <!-- Don't copy the reference assembly to output -->
+      <Private>false</Private>
+      <UndefineProperties>OSGroup</UndefineProperties>
+    </ProjectReference>
     <ProjectReference Include="..\src\System.Runtime.csproj">
+      <Project>{1e689c1b-690c-4799-bde9-6e7990585894}</Project>
+      <Name>System.Runtime.CoreCLR</Name>
```
Comment (weshaggard) [inline src/System.Runtime/tests/System.Runtime.Tests.csproj]: Why the name System.Runtime.CoreClr?

Comment (pallavit) [inline src/System.Runtime/tests/System.Runtime.Tests.csproj]: No particular reason. I was just trying to be consistent with the rest of the repo. I saw a couple of other libraries and each was following the same pattern.

Comment (weshaggard) [inline src/System.Runtime/tests/System.Runtime.Tests.csproj]: There shouldn't be any more .CoreCLR extensions for these projects any more, so please not add it here. That said the Name metadata itself also isn't needed really.

**Comment by weshaggard** [comment]
LGTM

File: src/System.Runtime/tests/System.Runtime.Tests.csproj (hunk 2 above)
Comment (pallavit) [inline src/System.Runtime/tests/System.Runtime.Tests.csproj]: Sounds good. Why were they added initially? I did not see anyone consuming the Name metadata but just let it be.

Comment (weshaggard) [inline src/System.Runtime/tests/System.Runtime.Tests.csproj]: I think VS might use it but it isn't required as VS will default to the name of the project as the Name.

**Comment by pallavit** [comment]
@weshaggard Could you please take a look at the Activator.CreateInstance commit.

**Comment by pallavit** [comment]
Please don't merge I need to make a parallel change in TFS for baseline comparison.

**Comment by weshaggard** [comment]
LGTM

**Comment by pallavit** [comment]
Thanks for the review.

**Pull request merged**